src/buckets/pagination/BucketList.tsx

```diff
@@ -0,0 +1,138 @@
+// Libraries
+import React, {PureComponent, RefObject, createRef} from 'react'
+import {withRouter, RouteComponentProps} from 'react-router-dom'
+import memoizeOne from 'memoize-one'
+
+// Components
+import BucketCard from 'src/buckets/components/BucketCard'
+import DemoDataBucketCard from 'src/buckets/components/DemoDataBucketCard'
+import {ResourceList} from '@influxdata/clockface'
+
+// Selectors
+import {getSortedResources} from 'src/shared/utils/sort'
+
+// Types
+import {Bucket, OwnBucket, Pageable} from 'src/types'
+import {PaginationNav, Sort} from '@influxdata/clockface'
+
+// Utils
+import {SortTypes} from 'src/shared/utils/sort'
+
+interface Props {
+  buckets: Bucket[]
+  bucketCount: number
+  emptyState: JSX.Element
+  onUpdateBucket: (b: OwnBucket) => void
+  onDeleteBucket: (b: OwnBucket) => void
+  onGetBucketSchema: (b: OwnBucket) => void
+  onFilterChange: (searchTerm: string) => void
+  pageHeight: number
+  pageWidth: number
+  sortKey: string
+  sortDirection: Sort
+  sortType: SortTypes
+}
+
+const DEFAULT_PAGINATION_CONTROL_HEIGHT = 62
```

Comment:
this is duplicated in a couple of places and will need to be moved, probably into Clockface